src/Cli.Tests/ValidateConfigTests.cs

```diff
@@ -46,6 +46,27 @@ public void TestConfigWithCustomPropertyAsInvalid()
         Assert.IsFalse(isConfigValid);
     }
 
+    /// <summary>
+    /// This method validates that the correct error messaging occurs when a config
+    /// with an invalid connection string is used that has relationship fields defined.
+    /// This case is a regression test due to the metadata needed not always being
+    /// populated in the SqlMetadataProvider if for example a bad connection string
+    /// is given. The error messages occur then when functions that depend on this
+    /// metadata are called.
+    /// </summary>
+    [TestMethod]
+    public void TestErrorMessageFromConfigWithNonWorkingConnectionString()
```

Comment:
Function name is too vague to represent the specific issue of validating relationship metadata when database metadata can't be fetched.

For the summary, do you mean: "This test ensures that relationship validation doesn't fail due to unhandled exceptions thrown as side effects of missing database metadata. A specific error message is expected which informs the user which entity/relationship caused validation to fail."?